resource.go

```diff
@@ -266,45 +326,13 @@ func (d ResourceDefinition) NeedUpdate(rs *ResourceStatus) bool {
 	return curImage != d.Image
 }
 
-func (d ResourceDefinition) rank() int {
-	switch d.Kind {
-	case "Namespace":
-		return 10
-	case "ServiceAccount":
-		return 20
-	case "CustomResourceDefinition":
-		return 30
-	case "ClusterRole":
-		return 40
-	case "ClusterRoleBinding":
-		return 50
-		// other cluster-scoped resources: 1000
-	case "Role":
-		return 2000
-	case "RoleBinding":
-		return 2010
-	case "NetworkPolicy":
-		return 2020
-	case "Secret":
-		return 2030
-	case "ConfigMap":
-		return 2040
-		// other namespace scoped resources: 3000
-	}
-
-	if d.Namespace == "" {
-		return 1000
-	}
-	return 3000
-}
-
 // SortResources sort resources as defined order of creation.
 func SortResources(res []ResourceDefinition) {
 	less := func(i, j int) bool {
 		a := res[i]
 		b := res[j]
-		aRank := a.rank()
-		bRank := b.rank()
+		aRank := a.Rank
+		bRank := b.Rank
 
```

Comment:
This change makes `Rank` a mandatory field. I suspect we need to update `/static/resources.go` and other stuffs?
https://github.com/cybozu-go/cke/blob/main/static/resources.go

Or we can use zero as a default value and resolve it here dynamically.
```go
aRank := a.Rank
if aRank == 0 {
  aRank = DecideRank(...)
}
```

Reply:
About static resources, we generate them by `compile_resources` in advance of starting cke.
And its output is sorted.
So we don't have to resolve it dynamically.
But I think that adding the rank field to the output of `compile_resources` is good.